consensus/istanbul/backend/announce.go

```diff
@@ -589,130 +649,311 @@ func (sb *Backend) regossipAnnounce(msg *istanbul.Message, payload []byte) error
 	return nil
```

Comment:
On line 638, unlocking before updating the time allows a race condition where any number of announce messages from the same address may be regossiped as long as they are received at the same time.